src/autorest.bicep/src/type-generator.ts

```diff
@@ -0,0 +1,451 @@
+
+    return output;
+  }
+
+  function* getDiscriminatedSubTypes(putSchema: ObjectSchema | undefined, getSchema: ObjectSchema | undefined) {
+    const putSubTypes = flattenDiscriminatorSubTypes(putSchema);
+    const getSubTypes = flattenDiscriminatorSubTypes(getSchema);
+
+    for (const subTypeName of uniq([...keys(putSubTypes), ...keys(getSubTypes)])) {
+      yield { 
+        subTypeName,
+        putSubType: putSubTypes[subTypeName],
+        getSubType: getSubTypes[subTypeName],
+      };
+    }
+  }
+
+  function parseType(putSchema: Schema | undefined, getSchema: Schema | undefined): TypeReference | undefined {
+    const combinedSchema = combineAndThrowIfNull(putSchema, getSchema);
+
+    // A schema that matches a JSON object with specific properties, such as
+    // { "name": { "type": "string" }, "age": { "type": "number" } }
+    if (combinedSchema instanceof ObjectSchema) {
+      return parseObjectType(putSchema as ObjectSchema, getSchema as ObjectSchema, true);
+    }
+
+    // A schema that matches a "dictionary" JSON object, such as
+    // { "additionalProperties": { "type": "string" } }
+    if (combinedSchema instanceof DictionarySchema) {
+      return parseDictionaryType(putSchema as DictionarySchema, getSchema as DictionarySchema);
+    }
+
+    // A schema that matches a single value from a given set of values, such as
+    // { "enum": [ "a", "b" ] }
+    if (combinedSchema instanceof ChoiceSchema) {
+      return parseEnumType(putSchema as ChoiceSchema, getSchema as ChoiceSchema);
+    }
+    if (combinedSchema instanceof SealedChoiceSchema) {
+      return parseEnumType(putSchema as SealedChoiceSchema, getSchema as SealedChoiceSchema);
+    }
+    if (combinedSchema instanceof ConstantSchema) {
+      return parseConstant(putSchema as ConstantSchema, getSchema as ConstantSchema);
+    }
+
+    // A schema that matches an array of values, such as
+    // { "items": { "type": "number" } }
+    if (combinedSchema instanceof ArraySchema) {
+      return parseArrayType(putSchema as ArraySchema, getSchema as ArraySchema);
+    }
+
+    // A schema that matches simple values, such as { "type": "number" }
+    if (combinedSchema instanceof PrimitiveSchema) {
+      return parsePrimaryType(putSchema as PrimitiveSchema, getSchema as PrimitiveSchema);
+    }
+
+    // The 'any' type
+    if (combinedSchema instanceof AnySchema) {
+      return factory.lookupBuiltInType(BuiltInTypeKind.Any);
+    }
+
+    logWarning(`Unrecognized property type: ${combinedSchema.type}. Returning 'any'.`);
+    return factory.lookupBuiltInType(BuiltInTypeKind.Any);
+  }
+
+  function parsePropertyFlags(putProperty: Property | undefined, getProperty: Property | undefined) {
+    let flags = ObjectPropertyFlags.None;
+
+    if (putProperty && putProperty.required) {
+      flags |= ObjectPropertyFlags.Required;
+    }
+
+    if (!putProperty || putProperty.readOnly) {
+      flags |= ObjectPropertyFlags.ReadOnly;
+    }
+
+    if (!getProperty) {
```

Comment:
Should we also handle `"secret": true` and `"x-ms-mutability": ["create", "update"]`?

Reply:
Good point. I'll raise an issue for this, as the intention with this PR is to reach parity with the current generator.

Reply:
Raised #240

Reply:
Sounds good.